corehq/apps/settings/tests/test_views.py

```diff
@@ -105,29 +107,39 @@ def _create_view_for_user(self, user):
 
 
 @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=True)
-class TwoFactorSetupView_FormKwargs_Tests(SimpleTestCase):
+class TwoFactorSetupView_RegisteredMethods_Tests(SimpleTestCase):
+
     @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=False)
-    def test_phone_methods_are_prohibited_when_settings_are_disabled(self):
+    def test_phone_methods_are_not_registered_when_settings_are_disabled(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
 
-    def test_phone_methods_are_allowed_when_user_belongs_to_messaging_domain(self):
+    def test_phone_methods_are_registered_when_user_belongs_to_messaging_domain(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertTrue(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator', 'sms', 'call'], [m.code for m in registry.get_methods()])
```

Comment:
Similar to the above, should this change to only assert that 'sms' and 'call' are in the returned methods?

Reply:
Same as above, reverted this change in 4ccbb9f9c74e